store/src/java/com/zimbra/cs/imap/ImapLoadBalancingMechanism.java

```diff
@@ -182,6 +182,8 @@ public Server getImapServerFromPool(HttpServletRequest httpReq, List<Server> poo
                 pool.sort(serverComparator);
                 int clientIpHash = InetAddress.getByName(httpReq.getHeader(CLIENT_IP)).hashCode();
                 int serverPoolIdx = clientIpHash % pool.size();
+                if (serverPoolIdx < 0)
+                    serverPoolIdx = -serverPoolIdx;
```

Comment:
Trav is it better get absolute value  of `serverPoolIdx` or `clientIpHash`? Or does it matter?